### DISCUSSION TITLE
Problem of very slow search

### DISCUSSION THREAD

**Discussion by caaptusss** [jellyfin/jellyfin#12444] [General]
Hello,

I have been using Jellyfin for years. It's running version 10.9.2 on a Debian 11 system. The setup is integrated into a VM with 28 cores, 64GB of memory, and a 500GB Samsung 990Pro SSD mounted in RAID 0.

Jellyfin hosts around 3,500 movies and 900 complete TV series.

Since several updates, the search function has become extremely slow, whether on the desktop interface or on the apps (tested on LG, Android). The result for the first letter typed can take up to 5 minutes to appear. You can imagine how long it takes to search for a complete word.

I don't see any logs indicating a problem. No warnings or errors are reported, including for slow responses (enabled in the admin console).

Do you have any idea how to fix this issue, please?

Thank you very much.

**Comment by verybadsoldier** [comment]
Might be similar to this:
https://github.com/jellyfin/jellyfin/issues/12228

**Comment by G2G2G2G** [comment]
an easy fix is to stop searching every key press and wait like 1 second after keys stop being pressed and then start a search.
Otherwise it ends up searching several times while you are typing and those results lag the interface and the server.
If you type your search somewhere else and paste it, it's 10x faster

or have an option where you must press "enter" after a search to start it

@joshuaboniface maybe consider it, or point me to the exact javascript file that takes the input of a typed button and sends it to the backend so I can setup some temporary ghetto waiting function that isn't so greedy

I doubt your fixes will speed it up unless you are truly redoing the entirety of how your database works because the querys are bad
